packages/checkpoint-dev/zarf.yaml

```diff
@@ -0,0 +1,83 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/defenseunicorns/zarf/main/zarf.schema.json
+
+kind: ZarfPackageConfig
+metadata:
+  name: k3d-core-slim-dev
+  description: "Rehydratable UDS K3d + UDS Core Slim (Istio, UDS Operator and Keycloak) Checkpoint"
+  authors: "Defense Unicorns - Product"
+  # x-release-please-start-version
+  version: "0.28.0"
+  # x-release-please-end
+
+variables:
+  - name: CLUSTER_NAME
+    description: "Name of the cluster"
+    default: "uds"
+
+  - name: K3D_EXTRA_ARGS
+    description: "Optionally pass k3d arguments to the default"
+    default: ""
+
+  - name: NGINX_EXTRA_PORTS
+    description: "Optionally allow more ports through Nginx (combine with K3D_EXTRA_ARGS '-p <port>:<port>@server:*')"
+    default: "[]"
+
+components:
+  - name: destroy-cluster
+    required: true
+    description: "Optionally destroy the cluster before creating it"
+    actions:
+      onDeploy:
+        before:
+          - cmd: |
+              echo "This package requires elevated permissions to create - requesting sudo (if paused enter password)"
+              sudo echo "got sudo! success!"
+          - cmd: k3d cluster delete ${ZARF_VAR_CLUSTER_NAME}
+            description: "Destroy the cluster"
+          - cmd: |
+              sudo rm -rf data
+
+  - name: create-cluster
+    required: true
+    description: "Create the K3d cluster w/UDS Core pre-installed"
+    files:
+      - source: uds-checkpoint.tar
+        target: uds-checkpoint.tar
+    actions:
+      onCreate:
+        before:
+          - cmd: ./checkpoint.sh
+        onSuccess:
+          - cmd: |
+              sudo rm -rf data uds-checkpoint.tar
+      onDeploy:
+        after:
+          - cmd: |
+              sudo tar --blocking-factor=64 -xpf uds-checkpoint.tar data
+              K8S_TOKEN="$(sudo cat $(pwd)/data/k3s_data/server/token)"
+
+              docker load -i data/uds-k3d-checkpoint-latest.tar
+
+              k3d cluster create \
+                -p "80:80@server:*" \
+                -p "443:443@server:*" \
+                --api-port 6550 \
+                --k3s-arg "--disable=traefik@server:*" \
+                --k3s-arg "--disable=metrics-server@server:*" \
+                --k3s-arg "--disable=servicelb@server:*" \
+                --k3s-arg "--disable=local-storage@server:*" \
+                --k3s-arg "--token=${K8S_TOKEN}@server:*" \
+                -v "$(pwd)/data/kubelet_data:/var/lib/kubelet@server:*" \
+                -v "$(pwd)/data/k3s_data:/var/lib/rancher/k3s@server:*" \
+                --image ghcr.io/defenseunicorns/uds-core/checkpoint:latest ${ZARF_VAR_K3D_EXTRA_ARGS} \
+                ${ZARF_VAR_CLUSTER_NAME}
+            description: "Create the cluster"
+          - description: Keycloak to be Healthy
+            wait:
+              cluster:
+                kind: Pod
+                name: app.kubernetes.io/name=keycloak
+                namespace: keycloak
+                condition: Ready
```

Comment:
This is really to wait for any Pod from the statefulset to be ready (since the Statefulset itself doesn't have a status) - KC is the last thing that will come up healthy and this helps the package mimic the Helm install so that downstream packages can interact with things immediately (without this wait there would be ~30s or so that you'd be waiting for KC to wake up)

Reply:
Most packages wouldn't _really_ need that and pepr should retry (and should be the only thing hitting KC directly off the jump) but I did it to be safe - I can remove it or add a comment if you'd like

Reply:
Comment might be nice - I assumed it was some sort of "readiness" but might not have understood that this is used as it's the last thing to come up (thinking in the future slim-dev will also include authservice, although Keycloak will likely still be the slowest piece).